File: app/src/main/java/com/example/mypc/fastfoodfinder/model/Routing/Route.java
package com.example.mypc.fastfoodfinder.model.Routing;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by nhoxb on 11/11/2016.
 */
public class Route implements Parcelable {
    protected Route(Parcel in) {
        legList = in.createTypedArrayList(Leg.CREATOR);
        summary = in.readString();
    }

    public static final Creator<Route> CREATOR = new Creator<Route>() {
        @Override
        public Route createFromParcel(Parcel in) {
            return new Route(in);
        }

        @Override
        public Route[] newArray(int size) {
            return new Route[size];
        }
    };

    public List<Leg> getLegList() {
        return legList;
    }

    public String getSummary() {
        return summary;
    }

    @SerializedName("legs")
    List<Leg> legList;
    @SerializedName("summary")
    String summary;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeTypedList(legList);
        parcel.writeString(summary);
    }
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/data/remote/routing/model/Route.kt
package com.iceteaviet.fastfoodfinder.data.remote.routing.model

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.JsonObject
import com.google.gson.JsonParser
import com.google.gson.annotations.SerializedName

/**
 * Created by Genius Doan on 11/11/2016.
 */
class Route constructor(`in`: Parcel) : Parcelable {

    @SerializedName("legs")
    val legList: List<Leg>
    @SerializedName("summary")
    val summary: String
    @SerializedName("overview_polyline")
    var encodedPolyline: JsonObject

    val encodedPolylineString: String
        get() = encodedPolyline.getAsJsonPrimitive("points").asString

    init {
        legList = `in`.createTypedArrayList(Leg.CREATOR)
        summary = `in`.readString()
        val parser = JsonParser()
        encodedPolyline = parser.parse(`in`.readString()).asJsonObject
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeTypedList(legList)
        parcel.writeString(summary)
        parcel.writeString(encodedPolyline.toString())
    }

    companion object CREATOR : Parcelable.Creator<Route> {
        override fun createFromParcel(parcel: Parcel): Route {
            return Route(parcel)
        }

        override fun newArray(size: Int): Array<Route?> {
            return arrayOfNulls(size)
        }
    }
}
